Source/UIImageExtension.swift

```diff
@@ -30,3 +48,41 @@ extension UIImage {
     return UIImage(data: data)?.size
   }
 }
+
+private extension CGSize {
+  /// Finds a new size constrained by a size keeping the aspect ratio.
+  ///
+  /// :param: size The contraining size.
+  /// :returns: size A new size that fits inside the contraining size with the same aspect ratio.
+  func sizeConstrainedBySize(size: CGSize) -> CGSize {
+    if height == 0 { return size }
+
+    let aspectRatio = width / height
+    let aspectWidth = round(aspectRatio * size.height)
+    let aspectHeight = round(size.width / aspectRatio)
+
+    if aspectWidth > size.width {
+      return CGSize(width: size.width, height: aspectHeight)
+    } else {
+      return CGSize(width: aspectWidth, height: size.height)
+    }
+  }
+
+  /// Finds a new size filling the given size while keeping the aspect ratio.
+  ///
+  /// :param: size The contraining size.
+  /// :returns: size A new size that fills the contraining size keeping the same aspect ratio.
+  func sizeFillingSize(size: CGSize) -> CGSize {
+    if height == 0 { return size }
+
+    let aspectRatio = width / height
```

Comment:
These three lines are repeated in both methods. Can we make them computed properties maybe?